### DISCUSSION TITLE
The curly brackets of the set are not rendered.

### DISCUSSION THREAD

**Discussion by zhenzhenChange** [community/community#47295] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

The curly brackets `{ }` of the set are not rendered.

Math Markdown Code

```text
$\{1, 2, 3\}$
```

VSCode

![image](https://user-images.githubusercontent.com/51586637/218611112-b107ddf9-c035-4706-baa5-c81614059892.png)

GitHub

![image](https://user-images.githubusercontent.com/51586637/218611083-88390fc5-3424-4eac-96bc-af774125f4b2.png)

**Comment by radar-blip** [comment, marked as answer]
Try like this: `$\\{1, 2, 3\\}$`

$\\{1, 2, 3\\}$

**Reply by queenofcorgis** [reply]
Nice - thanks for the solve @linuxlawson!

**Reply by radar-blip** [reply]
You're welcome!

**Comment by zhenzhenChange** [comment]
OK, thanks.

It's true that GitHub renders this way, but vscode fails.

![image](https://user-images.githubusercontent.com/51586637/218634601-fdf98dcf-db3f-4324-b92c-a4143c9de264.png)

Perhaps you are not using the same parsing algorithm?

**Reply by qijizh** [reply]
github may have escaped the markdown plugin before passing it to tex, whereas vscode passed it directly to tex.

**Reply by zhenzhenChange** [reply]
Yes, except that this difference can make it difficult to test for correctness because there is so much mathematical markup syntax.

It may have something to do with the preview plugin, I've now switched to [Markdown Preview Github Styling](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-preview-github-styles).

Input:

![image](https://user-images.githubusercontent.com/51586637/218906608-ae01d5b3-9a6b-46d2-9cd4-f886edad369a.png)


Output:

![image](https://user-images.githubusercontent.com/51586637/218906596-40993b93-2693-470d-bcb1-e0efe851b266.png)

But it still doesn't render correctly.

How can the rendering algorithm be consistent between vscode and GitHub? Based on a mathematical markdown syntax? Or could an official Markdown rendering plugin be provided?

**Comment by zhenzhenChange** [comment]
Link: https://github.com/mjbvz/vscode-github-markdown-preview-style/issues/99

**Comment by nschloe** [comment]
Perhaps a better fix for this is to use the new backtick-delimited inline math syntax:

```markdown
$`\{1, 2, 3\}`$
```

> $`\{1, 2, 3\}`$

You can use regular LaTeX syntax here.

**Reply by zhenzhenChange** [reply]
But it's not true for `VSCode`, the point here is whether you can find a generic way to be compatible with both `GitHub` and `VSCode`, with plugin support as well.

<img width="1203" alt="图片" src="https://github.com/community/community/assets/51586637/241aee69-ae68-4303-875f-acf613e0b27f">

**Reply by nschloe** [reply]
This is rather a shortcoming in VSCode; the backtick syntax is used by both GitHub and GitLab.

**Reply by samestep** [reply]
I opened a VS Code issue about this: microsoft/vscode#208430
